add TS offline mode (#159)
amplitude/amplitude-dev-center: docs/data/sdks/browser-2/index.md

  identityStorage: "localStorage",
});
```

### Offline mode

!!!note "Auto flush when reconnect"

>     Note that setting `config.flushIntervalMillis` to a very small value like 1 might cause `ERR_NETWORK_CHANGED`.

Reviewer: ```suggestion
    Setting `config.flushIntervalMillis` to a small value like `1` may cause an `ERR_NETWORK_CHANGED` error.
```

> Starting from version 2.4.0, the Amplitude Browser SDK supports offline mode. The SDK checks network connectivity every time it tracks an event. If the device is connected to network, the SDK schedules a flush. If not, it saves the event to storage. The SDK also listens for changes in network connectivity and schedules a flush based on `config.flushIntervalMillis` for all stored events when the device reconnects.

Reviewer: ```suggestion
Beginning with version 2.4.0, the Amplitude Browser SDK supports offline mode. The SDK checks network connectivity every time it tracks an event. If the device is connected to network, the SDK schedules a flush. If not, it saves the event to local storage. The SDK also listens for changes in network connectivity and schedules a flush of all stored events when the device reconnects, based on the `config.flushIntervalMillis` setting.
```

Is it accurate to specifically call out local storage here? I think we should be explicit that it's storing in the browser.

Author: No, local storage is the default option. But customers can use their custom storages too. So I'd keep it without storage here. But thanks for calling out!

Author: Updated with the rest of the suggest at https://github.com/amplitude/amplitude-dev-center/pull/159/commits/f0383800dd435502b9cc69578f410315229ce701 and https://github.com/amplitude/amplitude-dev-center/pull/159/commits/d4106d8dd7cadcf314c05675dd6172fd3d8b1d0a

> To opt-out, refer to the code below. Otherwise, you can omit the configuration to keep it on.

Reviewer: ```suggestion
To disable offline mode, add `offline: amplitude.Types.OfflineDisabled` to the `amplitude.init()` call as shown below.
```
